### ISSUE TITLE
Cannot resolve `tsx` components ?

### ISSUE THREAD

**Issue by Sliov** [souporserious/mdxts#77]
Hey there,

Awesome project, we're currently moving away from `contentlayer` and exploring multiple options, `mdxts` is quite simple and this is why we love it!

However, we're struggling a bit resolving `tsx` file like [mentioned in the docs](https://mdxts.dev/docs/getting-started#rendering-content-2).

 I spent around ~3 hours debugging it, cloned your repo and everything works fine when running the "site" folder, I tried to copy the dependencies & the `pnpm-lock.yaml` file but no luck with a fresh create-next-app 

Here's a repro
https://codesandbox.io/p/devbox/elegant-bhaskara-2fhqgd?file=%2Fapp%2Fcomponents%2F%5Bslug%5D%2Fpage.tsx%3A14%2C47

```data.ts
// data.ts
import { createSource } from "mdxts";

export const allDocs = createSource("docs/**/*.mdx");
export const allComponents = createSource("components/**/*.tsx");
```

```tsx
// app/components/[slug]/page.tsx
...

const component = await allComponents.get(`components/${params.slug}`);

console.log({ empty: await allComponents.all() });
// ^? {}

```

What am I doing wrong? 😕

**Comment by souporserious** [comment]
Hey, glad you're finding MDXTS simple so far, besides this issue 😄

My guess is this has to do with how "public" files are being calculated right now based on `package.json` exports [here](https://github.com/souporserious/mdxts/blob/main/packages/mdxts/src/utils/get-all-data.ts#L101-L109). This is why `site` is able to analyze the outside library's package exports, but assuming something broke with local package analysis.

I won't be able to take a closer look at this until the weekend, but I'll work on a fix as well as adding docs around this since it isn't mentioned anywhere. I appreciate you putting together a repro and filing an issue!

**Comment by souporserious** [comment]
Alright, got a chance to take a look at this! I was heading out of town right when I responded and wasn't exactly sure what was going on. So a few issues here that were causing your sandbox not to work:

- Currently an `index.(ts|tsx)` file is required to pick up the component exports as this heuristic is used to determine public exports. Adding an index file correctly picks up the `Button` component metadata.
- The link to the component was using an uppercase `Button` when it should be lowercase `button`

[Here's a working example](https://codesandbox.io/p/devbox/elegant-bhaskara-forked-5vchg5?file=%2Fdata.ts%3A6%2C29&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cltzbahzp00093b6j4twy3gn4%2522%252C%2522sizes%2522%253A%255B57.175324675324674%252C42.824675324675326%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cltzbahzp00033b6j7lz7dlyh%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cltzbahzp00073b6ju1m5zi04%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cltzbahzp00083b6jsl7dyr23%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B52.30333734756309%252C47.69666265243691%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cltzbahzp00033b6j7lz7dlyh%2522%253A%257B%2522id%2522%253A%2522cltzbahzp00033b6j7lz7dlyh%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltzbuwbr00023b6jpzsvbxih%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A6%252C%2522startColumn%2522%253A29%252C%2522endLineNumber%2522%253A6%252C%2522endColumn%2522%253A29%257D%255D%252C%2522filepath%2522%253A%2522%252Fdata.ts%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cltzbuwbr00023b6jpzsvbxih%2522%257D%252C%2522cltzbahzp00083b6jsl7dyr23%2522%253A%257B%2522id%2522%253A%2522cltzbahzp00083b6jsl7dyr23%2522%252C%2522activeTabId%2522%253A%2522clu4fxblq00ox3b6j30qh0zth%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522cltzbrfxt003v3b6ju4v6w50a%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clu4fxblq00ox3b6j30qh0zth%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252Fcomponents%252Fbutton%2522%257D%255D%257D%252C%2522cltzbahzp00073b6ju1m5zi04%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltzbahzp00043b6jbi9jxxma%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522cltzbahzp00053b6j4jzsj8ov%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522npm%2520install%2522%257D%252C%257B%2522id%2522%253A%2522cltzbahzp00063b6jw0egc4la%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clu4fy81p000eddfwb8j1bxzl%2522%257D%255D%252C%2522id%2522%253A%2522cltzbahzp00073b6ju1m5zi04%2522%252C%2522activeTabId%2522%253A%2522cltzbahzp00043b6jbi9jxxma%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

Again, I'll work on docs here so it's more clear how this currently works. As for how it _could_ work, would you prefer an option to preserve or change the casing for the generated routes so `components/Button` would work? And for the `index` file heuristic I think it'd make sense to fallback to reading the entire directory's exports if no `index` file is present so in your case here it would have been picked up, does that make sense?

**Comment by Sliov** [comment]
Makes sense, thank you for having a look at it!

I'm not too sure what's the best option here for preserving / changing the case, maybe introducing a formatter is probably too overkill?

Our use case is the following:

1. Define docs/button.mdx to write Button documentation – picked up by `createSource("docs/**/*.mdx", {});`
2. Define Button.tsx & Button.examples.tsx – picked up by `createSource("components/**/*.tsx", {});`
3. Render both in one page using `mergeSources` OR streaming the component route.

Ideally, if the [createSource options](`https://www.mdxts.dev/packages/mdxts#api-reference`) could provide a way to format the component name:

```ts
import { createSource } from 'mdxts'

export const allDocs = createSource('docs/**/*.mdx', {
  baseDirectory: 'docs',
})


export const allComponents = createSource('components/**/*.tsx', {
  format: (componentName) => changeCase.camelCase(componentName), // ==> 'button'
})

```

But we can handle this logic somewhere else when mergingSources, don't worry ✌️

**Comment by souporserious** [comment]
Ok, the missing `tsx` files are now fixed in https://github.com/souporserious/mdxts/pull/76 and released in version 0.10.0! I've also updated the documentation to detail how public metadata is calculated for JavaScript and TypeScript files. Please let me know if you run into any other troubles with that and I can take a look, thank you again for raising this issue!

Pertaining to the casing of the generated routes, I'd like to wait before adding an option too preemptively. If you can't work around it, I'd be happy to discuss an option there for custom formatting in a follow-up issue. I have some ideas to make `mergeSources` work better with external MDX docs that aren't colocated with the source code file so you don't need to stitch them together yourself.

**Comment by Sliov** [comment]
Amazing @souporserious , thank you so much 🙏
